### PR TITLE
Add MIT license to the repo.

### PR DESCRIPTION
**Pull request by Tzky**
I'm opening this as a PR to discuss the licensing of the repo. Personally i'd prefer to stick to the MIT license (that's what we usually use on the other subprojects), but i'd like to get some feedback first, before pulling this. So:

Thoughts?

Ping: @SpongePowered/developers 

EDIT:
We also need to get the consent of all current contributors, much like Forge did in https://github.com/MinecraftForge/MinecraftForge/issues/2789. (Thanks for pointing that out jamie)

### CODE REVIEW

File: controllers/announcements.go [hunk 1]
```diff
@@ -1,3 +1,26 @@
+// This file is part of SpongeHome, licensed under the MIT License (MIT).
+
```
Comment (jamierocks) [inline controllers/announcements.go]: Golang has block comments ;)

File: public/assets/js/spongehome.js [hunk 2]
```diff
@@ -1,3 +1,26 @@
+// This file is part of SpongeHome, licensed under the MIT License (MIT).
+
```
Comment (jamierocks) [inline public/assets/js/spongehome.js]: As does Javascript ;)

File: public/assets/scss/_base.scss [hunk 3]
```diff
@@ -1,3 +1,26 @@
+// This file is part of SpongeHome, licensed under the MIT License (MIT).
+
```
Comment (jamierocks) [inline public/assets/scss/_base.scss:2]: and SCCS :P

Comment (Tzky) [inline public/assets/scss/_base.scss:2]: I read about scss block comments staying in the actual css file when using block comments. That's why i chose normal comments.

Comment (jamierocks) [inline public/assets/scss/_base.scss:2]: Not sure if it does that with the partials, however that's definitely true for the main file.

File: license.txt [hunk 4]
```diff
@@ -0,0 +1,22 @@
+The MIT License (MIT)
+
```
Comment (jamierocks) [inline license.txt]: LICENSE.txt pls

File: public/assets/js/spongehome.js [hunk 5]
```diff
@@ -1,3 +1,28 @@
+/*
+This file is part of SpongeHome, licensed under the MIT License (MIT).
```
Comment (jamierocks) [inline public/assets/js/spongehome.js]:
``` js
/*
 * This file is part of SpongeHome, licensed under the MIT License (MIT).
 *
 */
```

pls :P

**Comment by JBYoshi** [comment]
> We also need to get the consent of all current contributors

That would be:
- [x] @jamierocks ([consent](https://github.com/SpongePowered/SpongeHome/pull/83#issuecomment-218381392))
- [x] @Tzky (assumed because he created this PR) 
- [x] @MichaelHillcox 
- [x] @Owexz 
- [x] @progwml6 ([consent](https://github.com/SpongePowered/SpongeHome/pull/83#issuecomment-218304548))
- [x] @simon816 

Feedback?

**Comment by progwml6** [comment]
The license file needs to be modified to be clear that this is just for our CSS/HTML/JS/GO code etc.
and not assets such as PNG.

the license on the HTML template files should probably be in non-rendered comments like this to reduce file sizes going to clients https://docs.djangoproject.com/en/dev/topics/templates/#comments

You have my permission to license any code/scripts i have made in this repo as MIT.

**Comment by jamierocks** [comment]
You have my permission to license my contributions to SpongeHome under the MIT license.

**Comment by Tzky** [comment]
"You have my permission to license my contributions to SpongeHome under the MIT license."

:)

**Comment by Tzky** [comment]
Made some changes to the license file... It now includes a statement about assets and about the truncate.js files.

**Comment by simon816** [comment]
Permission granted for having my contributions under the MIT license

**Comment by MichaelHillcox** [comment]
MIT licence is fine by me. You have my permission.

**Comment by progwml6** [comment]
@Tzky the package.json needs the license also

**Comment by Owexz** [comment]
I grant my permission for my contributions to SpongeHome to be subject to the MIT License.
Also FLARD.

**Comment by Tzky** [comment]
Ok, so every current contributor has agreed to the relicensing to MIT. Thus we're good to go on this PR.

Another question came to my mind: Shall we note (in the readme?) that every future contributor agrees to license his contributed code unter the MIT license? Or is this implied by our license file?

**Comment by progwml6** [comment]
There is no need to mention that they agree to the license once it is in master's license file... We might want to put a line in the readme file that this project is under the MIT license.

**Comment by jamierocks** [comment]
I'd just put:

The Sponge Project website. Licensed under the MIT license.

**Pull request merged**